nexus_constructor/main_window.py

```diff
@@ -208,14 +219,16 @@ def _load_current_entry():
 
     def set_button_state(self):
         indices = self.component_tree_view.selectedIndexes()
-        if len(indices) == 0 or len(indices) != 1:
+        if len(indices) != 1:
             self.delete_action.setEnabled(False)
             self.duplicate_action.setEnabled(False)
             self.new_rotation_action.setEnabled(False)
             self.new_translation_action.setEnabled(False)
             self.create_link_action.setEnabled(False)
+            self.zoom_action.setEnabled(False)
         else:
             selected_object = indices[0].internalPointer()
+            self.zoom_action.setEnabled(isinstance(selected_object, Component))
```

Comment:
The logic for when buttons are visible or not isn't tested, can we add some unit tests here please?

Reply:
The main window looks untestable for the time being...

Reply:
It still says blocked in #404.

Reply:
Ok, I think we need to do something about that, but it's outside the scope of this ticket. I've added a note to the linked ticket.